agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java

```diff
@@ -579,6 +584,66 @@ public void validate() throws FriendlyException {
         public GcReportingLevel reportingLevel = GcReportingLevel.TENURED_ONLY;
     }
 
+    public static class AadAuthentication {
+        public AadAuthentication() {
+            this.authenticationEnabled = false;
+        }
+        public boolean authenticationEnabled;
+        public AuthenticationType type;
+        public String clientId;
+        public String keePassDatabasePath;
+        public String tenantId;
+        public String clientSecret;
+        public String authorityHost;
```

Comment:
what is authorityHost? In applicationinsights_aadauth.json, it only lists type and clientId, how does customer know which type they need? Can we make type automatically configured? Which fields are optional? Can you document it?

Reply:
All this information can be found on the Azure identity documentation.